### PR TITLE
SignedTransaction and multi signature guide

### PR DESCRIPTION
**Pull request by sczembor**
Closes #x

## Proposed Changes

- Create `signedTransaction` guide for `Runtime`
- Create `multi signature` guide for `Runtime`

## TODO

- [x] update change log

### CODE REVIEW

File: docs/guide/signedTransaction.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+const signedTransacion = algosdk.decodeSignedTransaction(transaction.signTxn(alice.account.sk));
+// submit the transaction
+const confirmedTxn = runtime.executeTx([signedTransacion]);
+console.log(confirmedTxn);
+```
+Check the example in our [tests](../../packages/runtime/test/src/guide-examples.ts).
+
+# Multisignature
+
+Support of `SignedTransaction` objects in `Runtime` enables users to test their multisignatures. Again with the help of `algosdk` we can create multisignarue address, rekey an existing runtime account to this address and then sign transactions using multisignature accounts. `Runtime` verifies the signature and if its valid process the transaction.
```
Comment (Megha-Dev-19) [inline docs/guide/signedTransaction.md:38]:
```suggestion
Support of `SignedTransaction` objects in `Runtime` enables users to test their multisignatures. Again with the help of `algosdk` we can create multisignature address, rekey an existing runtime account to this address and then sign transactions using multisignature accounts. `Runtime` verifies the signature and if it's valid, it process the transaction.
```

File: packages/runtime/test/src/guide-examples.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	const fee = 1000;
+	let defaultBalance: bigint;
+	this.beforeEach(function () {
+		runtime = new Runtime([]);
+		[alice, bob, charlie, elon] = runtime.defaultAccounts();
+		defaultBalance = alice.balance();
+	});
+
+	it("Should send 5 algos from alice to bob", function () {
+		//create the transaction
```
Comment (Megha-Dev-19) [inline packages/runtime/test/src/guide-examples.ts:25]:
```suggestion
		// creates the transaction
```

File: packages/runtime/test/src/guide-examples.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		const transaction = algosdk.makePaymentTxnWithSuggestedParamsFromObject({
+			from: alice.address,
+			to: bob.address,
+			amount: amount,
+			note: undefined,
+			suggestedParams: mockSuggestedParams({ totalFee: fee }, runtime.getRound()),
+		});
+		// sign it and decode it to signedTransaction object
```
Comment (Megha-Dev-19) [inline packages/runtime/test/src/guide-examples.ts:33]:
```suggestion
		// signs and decode it to signedTransaction object
```

**Review by Megha-Dev-19 — COMMENTED** [review]
Left few suggestions

File: packages/runtime/src/ctx.ts [hunk 4]
```diff
@@ -253,7 +253,7 @@ export class Ctx implements Context {
 	 * @param flags asa deployment flags
 	 */
 	deployASA(name: string, fromAccountAddr: AccountAddress, flags: ASADeploymentFlags): ASAInfo {
-		if (this.runtime.loadedAssetsDefs === {}) {
+		if (this.runtime.loadedAssetsDefs === undefined) {
```
Comment (sczembor) [inline packages/runtime/src/ctx.ts:256]: not sure about this. My IDE warns me that the old expression always returns false

File: docs/guide/signedTransaction.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+## Example
+
+This example shows how to create a multisignature address using `algosdk`, rekey an existing runtime account to multisignautre, create a transaction and sign it using multisignature. For more details check the [algorand docs](https://developer.algorand.org/docs/get-details/transactions/signatures/#multisignatures).
```
Comment (thdailong) [inline docs/guide/signedTransaction.md:42]:
```suggestion
This example shows how to create a multisignature address using `algosdk`, rekey an existing runtime account to multisignature, create a transaction and sign it using multisignature. For more details check the [algorand docs](https://developer.algorand.org/docs/get-details/transactions/signatures/#multisignatures).
```

File: docs/guide/signedTransaction.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+const [alice, bob, charlie, elon] = runtime.defaultAccounts();
+// mock suggested params
+const suggestedParams = mockSuggestedParams({ totalFee: 1000 }, runtime.getRound());
+// create multisignature parameters
+const addrs = [bob.address, charlie.address];
+ 		multiSigParams = {
+ 			version: 1,
+ 			threshold: 2,
+ 			addrs: addrs,
+ 		};
```
Comment (thdailong) [inline docs/guide/signedTransaction.md:57]: The format here `multisigParams` look isn't good. Let change it

File: docs/guide/signedTransaction.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+// create multisignature address
+const multSigAddr = algosdk.multisigAddress(multiSigParams);
+// rekey alice to multi sig
```
Comment (thdailong) [inline docs/guide/signedTransaction.md:60]:
```suggestion
// rekey alice to multisignature account
```

File: docs/guide/signedTransaction.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+[alice, bob, charlie, elon] = runtime.defaultAccounts();
+// create transaction using algosdk
+const txn = algosdk.makePaymentTxnWithSuggestedParams(
+ 				alice.account.addr, // from
+ 				elon.account.addr, // to
+ 				5e6, // 5 algo
+ 				undefined,
+ 				undefined,
+ 				suggestedParams
+ 			);
```
Comment (thdailong) [inline docs/guide/signedTransaction.md:81]: Same here. Format isn't good

File: packages/runtime/test/src/guide-examples.ts [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+		// create multisignature parameters
+		const addrs = [bob.address, charlie.address];
+		const multiSigParams = {
+			version: 1,
+			threshold: 2,
+			addrs: addrs,
+		};
+		// create multisignature address
+		const multSigAddr = algosdk.multisigAddress(multiSigParams);
+		// rekey alice to multi sig
```
Comment (thdailong) [inline packages/runtime/test/src/guide-examples.ts:55]:
```suggestion
		// rekey alice to multisigature account
```

**Review by thdailong — COMMENTED** [review]
This looks good but you also need to change the format code in the guide.md

File: docs/guide/signedTransaction.md (hunk 8 above)
Comment (sczembor) [reply docs/guide/signedTransaction.md:81]: same as above

File: docs/guide/signedTransaction.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+const signedTransacion = algosdk.decodeSignedTransaction(transaction.signTxn(alice.account.sk));
+// submit the transaction
+const confirmedTxn = runtime.executeTx([signedTransacion]);
+console.log(confirmedTxn);
+```
+Check the example in our [tests](../../packages/runtime/test/src/guide-examples.ts).
+
+# Multisignature
+
+Support of `SignedTransaction` objects in `Runtime` enables users to test their multisignatures. Again with the help of `algosdk` we can create multisignature address, rekey an existing runtime account to this address and then sign transactions using multisignature accounts. `Runtime` verifies the signature and if it's valid, it process the transaction.
```
Comment (rmeena840) [inline docs/guide/signedTransaction.md:38]:
```suggestion
Support of `SignedTransaction` objects in `Runtime` enables users to test their multisignatures. Again with the help of `algosdk` we can create multisignature address, rekey an existing runtime account to this address and then sign transactions using multisignature accounts. `Runtime` verifies the signature and if it's valid, it executes the transaction.
```

File: docs/guide/signedTransaction.md [hunk 11]
```diff
@@ -0,0 +1,100 @@
+---
+layout: splash
+---
+
+# SignedTransaction
+
+`Runtime` can not only work with the `ExecParams` but also with `algosdk.SignedTransaction` objects. This allow for easier interaction with `algosdk` and provides more flexibility and functionality. We can create transactions and sign them using `algosdk` methods and execute them in `Runtime`.
```
Comment (robert-zaremba) [inline docs/guide/signedTransaction.md:7]:
```suggestion
`Runtime` supports transactions described as `ExecParams` and traditional `algosdk.SignedTransaction` objects. This allow for easier interaction with `algosdk` and provides more flexibility and functionality. We can create transactions and sign them using `algosdk` methods and execute them in `Runtime`. 
```

File: docs/guide/signedTransaction.md [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+layout: splash
+---
+
+# SignedTransaction
+
+`Runtime` can not only work with the `ExecParams` but also with `algosdk.SignedTransaction` objects. This allow for easier interaction with `algosdk` and provides more flexibility and functionality. We can create transactions and sign them using `algosdk` methods and execute them in `Runtime`. 
+
+## Example
+
+The example shows how to send 5 algo from one account to another using `Runtime`, `Runtime.defaultAccounts()` which are predefined, funded accounts and `algosdk` to create and sign the transaction. For more details check the [algorand docs](https://developer.algorand.org/docs/sdks/javascript/#build-first-transaction).
```
Comment (robert-zaremba) [inline docs/guide/signedTransaction.md:11]:
```suggestion
The example shows how to send 5 algo from one account to another using `Runtime`. We will use `Runtime.defaultAccounts()` which are predefined, funded accounts and `algosdk` to create and sign the transaction. For more details check the [algorand docs](https://developer.algorand.org/docs/sdks/javascript/#build-first-transaction).
```

File: docs/guide/signedTransaction.md (hunk 6 above)
Comment (robert-zaremba) [reply docs/guide/signedTransaction.md:57]: So why there is indentation?

File: docs/guide/signedTransaction.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+```ts
+// create runtime
+const runtime = new Runtime([]);
+// get two funded accounts
+const [alice, bob, charlie, elon] = runtime.defaultAccounts();
+// mock suggested params
+const suggestedParams = mockSuggestedParams({ totalFee: 1000 }, runtime.getRound());
+// create multisignature parameters
+const addrs = [bob.address, charlie.address];
+ 		multiSigParams = {
```
Comment (robert-zaremba) [inline docs/guide/signedTransaction.md:53]:
```suggestion
let multiSigParams = {
```

File: docs/guide/signedTransaction.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+    multiSigParams,
+    charlie.account.sk
+).blob;
+// decode the transaction
+const signedTxn: SignedTransaction = algosdk.decodeSignedTransaction(twosigs);
+// submit the transaction
+const confirmedTxn = runtime.executeTx([signedTxn]);
+console.log(confirmedTxn);
+```
+Check the example in our [tests](../../packages/runtime/test/src/guide-examples.ts).
```
Comment (robert-zaremba) [inline docs/guide/signedTransaction.md:100]:
```suggestion

### Full Code

Check the example in our [tests](../../packages/runtime/test/src/guide-examples.ts).
```

**Review by robert-zaremba — APPROVED** [review]

File: docs/guide/signedTransaction.md (hunk 8 above)
Comment (sczembor) [reply docs/guide/signedTransaction.md]: fixed it

File: docs/guide/signedTransaction.md (hunk 13 above)
Comment (sczembor) [reply docs/guide/signedTransaction.md]: fixed

File: docs/guide/signedTransaction.md (hunk 6 above)
Comment (sczembor) [reply docs/guide/signedTransaction.md]: I fixed it

**Review by thdailong — APPROVED** [review]
Forgot to approve! But now this one is good to me!
Also there is conflict you need to fix

**Pull request merged**